src/Microsoft.AspNet.Mvc.ModelBinding/Binders/CompositeModelBinder.cs

```diff
@@ -97,7 +97,17 @@ public CompositeModelBinder([NotNull] IEnumerable<IModelBinder> modelBinders)
                 var result = await binder.BindModelAsync(bindingContext);
                 if (result != null)
                 {
-                    return result;
+                    // Use returned ModelBindingResult if it either indicates model was set or is related to errors.
+                    // The second condition is necessary because some binders add errors that would be duplicated when
+                    // falling back to the empty prefix. Those binders often cannot run twice anyhow.
+                    if (result.IsModelSet || bindingContext.ModelState.ContainsKey(bindingContext.ModelName))
```

Comment:
this loop has always exited when it hit the first `binder` to return a non-`null` result.  all I've done is changed the method to return `null` in a few more cases, letting the caller fall back to the empty prefix.